### PR TITLE
Automatically flag reviews with links in the API

### PR DESCRIPTION
**Pull request by diox**
(As already done in the regular view)

Another little missing detail I noticed.

### CODE REVIEW

File: src/olympia/reviews/serializers.py [hunk 1]
```diff
@@ -34,6 +38,15 @@ def validate(self, data):
         # Also include the user ip adress.
         data['ip_address'] = request.META.get('REMOTE_ADDR', '')
 
+        # Clean up body and automatically flag the review if an URL was in it.
+        # Unquote the body in case someone tries 'example%2ecom'.
+        body = unquote(data.get('body', ''))
+        if '<br>' in data:
+            self.cleaned_data['body'] = re.sub('<br>', '\n', body)
```
Comment (EnTeQuAk) [inline src/olympia/reviews/serializers.py]: that's a very naive cleanup, django does have helpers that do that in a more reasonable fashion, maybe we should use those?

Comment (EnTeQuAk) [inline src/olympia/reviews/serializers.py]: And there's no test for that :)

Comment (diox) [inline src/olympia/reviews/serializers.py]: Note that it replaces `<br>` with `\n`, not the other way around. It's shamelessly copy/pasted from the original form. I'd be happy to use something else, but I don't think we care that much

Comment (diox) [inline src/olympia/reviews/serializers.py]: > And there's no test for that :)

shame

I'll add one.

Comment (diox) [inline src/olympia/reviews/serializers.py]: Now tested.

**Comment by eviljeff** [comment]
I can't help thinking this should be with the model in /reviews rather than in serializers, or views (or wherever the existing logic is).

**Comment by eviljeff** [comment]
r+ otherwise

**Comment by diox** [comment]
It makes sense in the model, and I almost did this, but it would require more refactoring with the existing views, and I don't really want to touch that mess... Since the serializer is responsible for transforming input into a model instance it's the next logical place to put that code, IMHO. It allows it to be re-used more than if it was done in the view.

**Pull request merged**